src/test/java/cohen/wordle/WordleControllerTest.java

```diff
@@ -0,0 +1,158 @@
+package cohen.wordle;
+
+import org.junit.jupiter.api.Test;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.ArrayList;
+
+import static cohen.wordle.CharStatus.*;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.mockito.Mockito.*;
+
+class WordleControllerTest
+{
+    WordleGame wordleGame = mock();
+    JLabel[][] letters = new JLabel[][]
+            {
+                    {mock(), mock(), mock(), mock(), mock()},
+                    {mock(), mock(), mock(), mock(), mock()},
+                    {mock(), mock(), mock(), mock(), mock()},
+                    {mock(), mock(), mock(), mock(), mock()},
+                    {mock(), mock(), mock(), mock(), mock()},
+                    {mock(), mock(), mock(), mock(), mock()}
+            };
+
+    @Test
+    public void addLetter()
+    {
+        //given
+        WordleController controller = new WordleController(wordleGame, letters);
+
+        //when
+        controller.addLetter("t");
+        controller.addLetter("a");
+        controller.addLetter("u");
+
+        //then
+        verify(letters[0][0]).setText("T");
+        verify(letters[0][1]).setText("A");
+        verify(letters[0][2]).setText("U");
+    }
+
+    @Test
+    public void backspaceAfterNone()
+    {
+        //given
+        WordleController controller = new WordleController(wordleGame, letters);
+
+        //when
+        controller.backspace();
+
+        //then
+        verifyNoInteractions(wordleGame);
+    }
+
+    @Test
+    public void backspaceAfterOne()
+    {
+        //given
+        WordleController controller = new WordleController(wordleGame, letters);
+
+        //when
+        controller.addLetter("t");
+        controller.backspace();
+        controller.addLetter("s");
+
+
+        //then
+        verify(letters[0][0]).setText("T");
+        verify(letters[0][0]).setText("");
+        verify(letters[0][0]).setText("S");
+    }
+
+    @Test
+    public void backspaceAfterMultiple()
+    {
+        //given
+        WordleController controller = new WordleController(wordleGame, letters);
+
+        //when
+        controller.addLetter("t");
+        controller.addLetter("a");
+        controller.backspace();
+        controller.addLetter("u");
+
+        //then
+        verify(letters[0][0]).setText("T");
+        verify(letters[0][1]).setText("A");
+        verify(letters[0][1]).setText("");
+        verify(letters[0][1]).setText("U");
+    }
+
+    @Test
+    public void enterValidGuess()
+    {
+        //given
+        WordleGame game = mock();
+        WordleController controller = new WordleController(game, letters);
+        CharStatus[] results = {NotFound, NotFound, NotFound, WrongPlace, Correct};
```

Comment:
Why are you retunring NotFound when it's the correct word?

Reply:
I want it to be that the word was not guessed, so it returns these results as if someone had guessed "phone" for example, instead of taupe

Reply:
Then you should set up the test that way.